### DISCUSSION TITLE
give role command

### DISCUSSION THREAD

**Discussion by AmirhN1382** [Rapptz/discord.py#6399] [Q&A] [answered]
```
@client.command(pass_context=True)
async def role(ctx, user=None, role=None):
    if not user:
        await ctx.send("Provide user")
        return
    if not role:
        await ctx.send("Provide Role")
        return
    role = discord.utils.get(ctx.message.author.guild.roles, name=role)
    print(f"User= {user} | Role= {role}")
    await user.add_roles(user, role, reason=f"used ^role by {ctx.message.author.name}")
    await ctx.send("DONE")
```
I don't know why but this is not working.
error: `AttributeError: 'str' object has no attribute 'add_roles'`

**Comment by tk421bsod** [comment, marked as answer]
`user` is a string. If you want to convert that argument to a Member, use a [typehint](https://stackoverflow.com/questions/32557920/what-are-type-hints-in-python-3-5) to discord.Member. (`user:discord.Member`) This will attempt to convert `user` to an instance of discord.Member, and this requires the members intent to be enabled, read the [docs for intents](https://discordpy.readthedocs.io/en/latest/intents.html) for more information.

**Reply by AmirhN1382** [reply]
i have enabled member intents, how should i use `user:discord.Member` ?

**Reply by bandwiches** [reply]
`async def role(ctx, user: discord.Member = None, role=None)`

**Reply by AmirhN1382** [reply]
worked! thanks

**Reply by Harmon758** [reply]
See [the Discord Converters section of the documentation](https://discordpy.readthedocs.io/en/latest/ext/commands/commands.html#discord-converters).

**Reply by Tapicz** [reply]
Does anyone know what i have to do, to execute this command from terminal?

**Reply by Harmon758** [reply]
I'm not sure what you're referring to, but it seems unlikely that it's related to this question.
